examples/math-c/lib/main.h

```diff
@@ -0,0 +1,103 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#define N 1000
+#define D 100000000
+
+// Iteratively adds the product of sine(x + i) and cosine(y + i) for i from 1 to 1,000,000.
+
+double itt_add()
+{
+    double x = 1.0;
+    double y = 2.0;
+    double result = 0.0;
+    int i;
+
+    for (i = 1; i <= D; i++) {
+        result += sin(x + i) * cos(y + i);
+    }
+
+    printf("%d\n",result);
+}
+
+
+
+
+void matrix_multiply()
+{
+    double **A, **B, **C;
+    int i, j, k;
+
+    // Allocate memory for matrices A, B, and C
+    A = (double **)malloc(N * sizeof(double *));
+    B = (double **)malloc(N * sizeof(double *));
+    C = (double **)malloc(N * sizeof(double *));
+    for (i = 0; i < N; i++) {
+        A[i] = (double *)malloc(N * sizeof(double));
+        B[i] = (double *)malloc(N * sizeof(double));
+        C[i] = (double *)malloc(N * sizeof(double));
+    }
+
+    // Initialize matrices A and B with random values
+    for (i = 0; i < N; i++) {
+        for (j = 0; j < N; j++) {
+            A[i][j] = rand() % 10;
+            B[i][j] = rand() % 10;
+        }
+    }
+
+    // Perform matrix multiplication
+    for (i = 0; i < N; i++) {
+        for (j = 0; j < N; j++) {
+            C[i][j] = 0;
+            for (k = 0; k < N; k++) {
+                C[i][j] += A[i][k] * B[k][j];
+            }
+        }
+    }
+
+    // Print the resulting matrix
+    printf("Resulting matrix:\n");
+    for (i = 0; i < N; i++) {
+        for (j = 0; j < N; j++) {
+            printf("%f ", C[i][j]);
+        }
+        printf("\n");
+    }
+
+    // Free the allocated memory
+    for (i = 0; i < N; i++) {
+        free(A[i]);
+        free(B[i]);
+        free(C[i]);
+    }
+    free(A);
+    free(B);
+    free(C);
+}
+
+
+double array_sum()
+{
+    double *arr;
+    int i;
+    double sum = 0.0;
+
+    // Allocate memory for the array
+    arr = (double *) malloc(D * sizeof(double));
+
+    // Initialize the array with random values
+    for (i = 0; i < D; i++) {
+        arr[i] = rand() % 10;
+    }
+
+    // Calculate the sum of the array
+    for (i = 0; i < D; i++) {
+        sum += arr[i];
+    }
+
+    free(arr);
+
+    printf("%d\n", sum);
+}
```

Comment:
Do we really need it? It shows C performance, not sushi performance.

Reply:
No, we don't need it. The goal was to write the equivalent code from C in Python to show performance. We can remove this

Reply:
I can submit a separate PR cleaning up this C code, removing the equivalent Python functions and solely focussing on sushi performance.

Reply:
Sure. Should I leave this example or remove it and wait until the another pull request?

Reply:
Can you leave this example? I will submit a different PR simplifying it. It will make discussion easier when focussing on one change.